spec/controllers/tasks_controller_spec.rb

```diff
@@ -252,6 +278,16 @@
         expect { put_update }.to change { task.reload.license.name }.to('new_license')
       end
 
+      it 'updates the tasks labels' do
+        expect { put_update }.to change { task.reload.labels }.from([existing_label]).to([new_label])
+      end
+
+      it 'creates new task label when requested' do
+        changed_attributes[:label_names] = ['some new label']
+        expect { put_update }.to change { task.reload.labels.map(&:name) }.from([existing_label.name]).to(['some new label'])
+        expect(Label.where(name: 'some new label')).not_to be_empty
+      end
```

Comment:
since you already checked that the update works in the example before I would just add the following
```suggestion
      it 'creates new task label when requested' do
        expect { put_update }.to change(Label, :count).by(1)
      end
```

Reply:
I moved the test into a new context. Maybe that is better.